### PR TITLE
Media Page and More More More Search fixes

### PR DESCRIPTION
**Pull request by Mikunj**
**Changes:**

- Improved media page load times
    - Media pages will now load the basic media data and then lazy load episodes/chapters, related media, reactions and characters.
- Updated some minor text in media pages
    - I.e showing Chapters instead of Episodes when media is `manga`
- Tab bar in media pages/profile pages now uses a horizontal scroll view.
    - This change is for devices with small screens, before all the tabs used to be squished together.
- Updated item layouts in media/profile pages.
- Minor bug fixes (look below for the things fixed).

**TODO:**

- [x] Upon initial load of search, The view width doesn't conform to the screen width and thus `View All` button and the Search bar aren't layed out properly. This however fixes itself after a few seconds but it would be better if it never did it in the first place. Find a fix/workaround for this?
- [x] Tapping View All Reactions in media pages on android causes a crash.
- [x] Keyboard should be dismissed upon scroll/submit when searching for anime/manga
- [x] Reactions on the summary page don't display properly if media name is too long (Fullmetal Alchemist: Brotherhood)
- [x] Tab bar for `Summary, Episodes, etc..` is squashed on devices with small screens (iphone 5s)
- [x] Tapping More from this series should take the user to the media page. (Maybe also show the media type on the item itself so users don't get confused? I.e Manga, OVA, TV etc..)
- [x] Update More from this series item layout to look closer to default search item layout.
- [ ] In Manga pages, Chapters should be shown instead of Episodes. It also crashes when you tap the episodes tab.
- [ ] Search results should show a loading indicator when loading more data (I.e when you scroll down all the way)
- [ ] Media page still shows `Add to library` even when it is in the user library??
- [ ] Reactions tab should infinite scroll?
- [ ] Episodes should show the title + their number

### CODE REVIEW

File: src/screens/Profiles/MediaPages/index.js [hunk 1]
```diff
@@ -71,49 +73,81 @@ class MediaPages extends PureComponent {
...
+    Kitsu.one(type, id).get({
+      include: `mediaRelationships.destination,${type === 'anime' ? 'episodes' : 'chapters'}`,
+    })
+      .then((media) => {
+        // Combine the 2 object that we have
+        this.setState({
+          media: { ...media, categories: this.state.media.categories },
+          loadingAdditional: false,
+        });
+      });
```
Comment (edencakir) [inline src/screens/Profiles/MediaPages/index.js]: Consider using `await` instead. It's really common in our codebase

File: src/screens/Profiles/MediaPages/pages/Summary.js [hunk 2]
```diff
@@ -109,24 +132,33 @@ class SummaryComponent extends PureComponent {
...
+            </ScrollItem>);
+          }}
         />
 
         {/* Reactions */}
         <ScrollableSection
           title="Reactions"
           onViewAllPress={() => this.navigateTo('Reactions')}
           data={mediaReactions}
+          loading={mediaReactions === null}
```
Comment (edencakir) [inline src/screens/Profiles/MediaPages/pages/Summary.js]: Use `lodash` https://lodash.com/docs/#isNull

File: src/screens/Profiles/MediaPages/pages/Summary.js [hunk 3]
```diff
@@ -144,9 +176,12 @@ class SummaryComponent extends PureComponent {
           title="Characters"
           onViewAllPress={() => this.navigateTo('Characters')}
           data={castings}
+          loading={castings === null}
```
Comment (edencakir) [inline src/screens/Profiles/MediaPages/pages/Summary.js]: same here

File: src/screens/Profiles/components/AvatarHeader/component.js [hunk 4]
```diff
@@ -25,11 +25,13 @@ AvatarHeader.propTypes = {
...
   sideElement: PropTypes.element,
+  boxed: PropTypes.bool,
 };
 
 AvatarHeader.defaultProps = {
   avatar: null,
   title: '',
   subtitle: '',
   sideElement: null,
+  boxed: false,
```
Comment (edencakir) [inline src/screens/Profiles/components/AvatarHeader/component.js]: What about making `boxed: undefined` and `numberOfLines={boxed && 1}` or something similar.

File: src/screens/Profiles/components/ImageCard/component.js [hunk 5]
```diff
@@ -1,63 +1,123 @@
...
+  thumbnail: 10,
+};
+
+const TextView = ({ variant, title, subtitle, boldTitle, centerTitle, noMask }) => {
+  const titleSize = variant === 'landscapeLarge' ? 'default' : 'xsmall';
+  const subtitleSize = variant === 'landscapeLarge' ? 'small' : 'xxsmall';
+
+  return (title || subtitle) && (
+    <View style={[styles.anchorBottom, { height: '60%' }]}>
+      { !noMask && <LinearGradient colors={['transparent', 'rgba(0,0,0,0.9)']} style={styles.linearGradient} />}
```
Comment (edencakir) [inline src/screens/Profiles/components/ImageCard/component.js]: Let's also define PropTypes & default props for `TextView`. Accidentally passing `String` to `noMask` will cause an error where debug line info is absent which probably cost us hours. Just a precaution.

File: src/screens/Profiles/components/ImageCard/component.js [hunk 6]
```diff
@@ -1,63 +1,123 @@
...
+      <View style={[styles.anchorBottom, { padding: paddingOptions[variant] }]}>
+        {subtitle && <StyledText color="lightGrey" size={subtitleSize} numberOfLines={1}>{subtitle}</StyledText>}
```
Comment (edencakir) [inline src/screens/Profiles/components/ImageCard/component.js]: try passing an empty string to the subtitle, it should crash the app. exactly the same reason with the above

File: src/screens/Profiles/components/ImageCard/component.js [hunk 7]
```diff
@@ -1,63 +1,123 @@
...
+        {title &&
```
Comment (edencakir) [inline src/screens/Profiles/components/ImageCard/component.js]: try passing an empty string to the title, it should crash the app. exactly the same reason with the above

File: src/screens/Profiles/components/ReactionBox/component.js [hunk 8]
```diff
@@ -21,9 +21,10 @@ export const ReactionBox = ({ boxed, reactedMedia, reaction }) => {
         title={reactedMedia}
         subtitle={`by ${user.name} ・ ${timeStamp}`}
         sideElement={<UpVoteCountBox upVotesCount={reaction.upVotesCount} />}
+        boxed={boxed}
       />
       <View style={styles.main}>
-        <StyledText color="dark" size="small" numberOfLines={boxed && 3}>{reaction.reaction}</StyledText>
+        <StyledText color="dark" size="small" numberOfLines={(boxed && 3) || undefined}>{reaction.reaction}</StyledText>
```
Comment (edencakir) [inline src/screens/Profiles/components/ReactionBox/component.js]: Same with the above

File: src/screens/Profiles/components/ReactionBox/styles.js [hunk 9]
```diff
@@ -9,7 +9,7 @@ export const styles = StyleSheet.create({
   },
   wrap__boxed: {
     borderRadius: 6,
-    width: cardSize.landscapeLarge.width,
+    width: Math.max(cardSize.landscapeLarge.width, 280),
```
Comment (edencakir) [inline src/screens/Profiles/components/ReactionBox/styles.js]: maybe `maxWidth: 280` in addition to line 12. But both are okay

File: src/screens/Profiles/components/TabHeader/component.js [hunk 10]
```diff
@@ -14,7 +14,7 @@ export const TabHeader = ({
 }) => (
   <View style={styles.wrap}>
     <TitleText color={contentDark ? 'dark' : 'lightGrey'}>{title}</TitleText>
-    {(actionTitle && actionOnPress) && (
+    {actionTitle && (
```
Comment (edencakir) [inline src/screens/Profiles/components/TabHeader/component.js]: try passing an empty string to the actionTitle, it should crash the app. exactly the same reason with the above

**Comment by edencakir** [comment]
This is how passing `''` error looks like:

File: src/screens/Profiles/components/ReactionBox/styles.js (hunk 9 above)
Comment (Mikunj) [reply src/screens/Profiles/components/ReactionBox/styles.js]: Woops, forgot to reset this when i made cardSizes constants.

**Pull request merged**